internal/cluster/validator.go

```diff
@@ -528,177 +379,127 @@ func checkCidrsOverlapping(cluster *common.Cluster) error {
 		return clusterNetwork != nil &&
 			network.VerifyNetworkHostPrefix(clusterNetwork.HostPrefix) == nil &&
 			network.VerifyClusterCidrSize(int(clusterNetwork.HostPrefix), string(clusterNetwork.Cidr), len(c.cluster.Hosts)) == nil
 	}).([]*models.ClusterNetwork)
 
-	return boolToValidationStatus(len(validClusterNetworks) == len(c.cluster.ClusterNetworks))
-}
-
-func (v *clusterValidator) printNetworkPrefixValid(c *clusterPreprocessContext, status ValidationStatus) string {
-	switch status {
-	case ValidationSuccess:
-		return "The Cluster Network prefix is valid."
-	case ValidationFailure:
-		var err error
+	if len(validClusterNetworks) == len(c.cluster.ClusterNetworks) {
+		return ValidationSuccess, "The Cluster Network prefix is valid."
+	}
 
-		for _, clusterNetwork := range c.cluster.ClusterNetworks {
-			if err = network.VerifyNetworkHostPrefix(clusterNetwork.HostPrefix); err != nil {
-				return fmt.Sprintf("Invalid Cluster Network prefix: %s.", err.Error())
-			} else if err = network.VerifyClusterCidrSize(int(clusterNetwork.HostPrefix), string(clusterNetwork.Cidr), len(c.cluster.Hosts)); err != nil {
-				return err.Error()
-			}
+	var err error
+	for _, clusterNetwork := range c.cluster.ClusterNetworks {
+		if err = network.VerifyNetworkHostPrefix(clusterNetwork.HostPrefix); err != nil {
+			return ValidationFailure, fmt.Sprintf("Invalid Cluster Network prefix: %s.", err.Error())
+		}
+		if err = network.VerifyClusterCidrSize(int(clusterNetwork.HostPrefix), string(clusterNetwork.Cidr), len(c.cluster.Hosts)); err != nil {
+			return ValidationFailure, err.Error()
 		}
-
-		return ""
-	case ValidationPending:
-		return "The Cluster Network CIDR is undefined."
-	default:
-		return fmt.Sprintf("Unexpected status %s", status)
 	}
+	return ValidationError, "Unexpected status ValidationError"
 }
 
-func (v *clusterValidator) isNtpServerConfigured(c *clusterPreprocessContext) ValidationStatus {
+func (v *clusterValidator) isNtpServerConfigured(c *clusterPreprocessContext) (ValidationStatus, string) {
 	synced, err := common.IsNtpSynced(c.cluster)
 	if err != nil {
-		return ValidationError
+		return ValidationError, fmt.Sprintf("Unexpected error %s", err.Error())
 	}
-	return boolToValidationStatus(synced)
-}
-
-func (v *clusterValidator) printNtpServerConfigured(c *clusterPreprocessContext, status ValidationStatus) string {
-	switch status {
-	case ValidationSuccess:
-		return "No ntp problems found"
-	case ValidationFailure:
-		return fmt.Sprintf("Hosts' clocks are not synchronized (there's more than a %d minutes gap between clocks), "+
-			"please configure an NTP server via DHCP or set clocks manually.", common.MaximumAllowedTimeDiffMinutes)
-	default:
-		return fmt.Sprintf("Unexpected status %s", status)
+	if synced {
+		return ValidationSuccess, "No ntp problems found"
 	}
+	return ValidationFailure, fmt.Sprintf("Hosts' clocks are not synchronized (there's more than a %d minutes gap between clocks), "+
```

Comment:
Small nit, but whenever possible I think the success case should be the least indented. So I'd reverse this.
```suggestion
	if !synced {
		return ValidationFailure, fmt.Sprintf("Hosts' clocks are not synchronized (there's more than a %d minutes gap between clocks), "+
	}
	return ValidationSuccess, "No ntp problems found"
```